### PR TITLE
Better autovectorization of `spatiotemporal_scale`

### PR DESCRIPTION
**Pull request by redzic**
It turns out that avoiding a call to `Iterator::take` allows LLVM to autovectorize `spatiotemporal_scale`.

For a comparison of the code generated (simplified for example): https://godbolt.org/z/56Kec186G

### CODE REVIEW

**Review by barrbrain — COMMENTED** [review]
I compared the generated code from this change against just removing the `.take(MAX_SB_IN_IMP_B)`. Both resulted in vectorised loops but this change was better because the output included such a loop for widths narrower than the maximum.

**Comment by redzic** [comment]
For me it was never vectorized without this change, when compiling for znver1. Also the CI errors look unrelated to the change.

**Comment by barrbrain** [comment]
The CI errors should be resolved by #2848. Rebasing this PR will apply that fix.

File: src/rdo.rs [hunk 1]
```diff
@@ -720,14 +720,33 @@ pub fn spatiotemporal_scale<T: Pixel>(
...
+      &slice[..n]
+    }
+  }
+
   let mut sum = 0;
   for y in y0..y1 {
-    sum += fi.distortion_scales[y * fi.w_in_imp_b..][x0..x1]
-      .iter()
-      .zip(fi.activity_scales[y * fi.w_in_imp_b..][x0..x1].iter())
-      .take(MAX_SB_IN_IMP_B)
```
Comment (barrbrain) [inline src/rdo.rs:728]: I checked again with `RUSTFLAGS="-C target-feature=+avx2,+fma" cargo asm -- rav1e::rdo::spatiotemporal_scale`.
Removing `.take(MAX_SB_IN_IMP_B)` here does result in auto-vectorisation but only for the maximum width.

**Review by barrbrain — APPROVED** [review]
This is better than the alternative of removing `.take()` as more widths receive auto-vectorised loops.

**Pull request merged**